x-pack/spec/filters/geoip/download_manager_spec.rb

```diff
@@ -0,0 +1,148 @@
+      before(:each) do
+        expect(download_manager).to receive(:get_uuid).and_return(SecureRandom.uuid)
+        mock_resp = double("geoip_endpoint", :body => ::File.read(::File.expand_path("./fixtures/normal_resp.json", ::File.dirname(__FILE__))), :status => 200)
+        allow(download_manager).to receive_message_chain("rest_client.get").and_return(mock_resp)
+      end
+
+      it "should return has_update and db info when md5 does not match" do
+        expect(mock_metadata).to receive(:gz_md5).and_return("")
+
+        has_update, info = download_manager.send(:check_update)
+        expect(has_update).to be_truthy
+        expect(info).to have_key("md5_hash")
+        expect(info).to have_key("name")
+        expect(info).to have_key("provider")
+        expect(info).to have_key("updated")
+        expect(info).to have_key("url")
+        expect(info["name"]).to include("City")
+      end
+
+      it "should return false when md5 is the same" do
+        expect(mock_metadata).to receive(:gz_md5).and_return("4013dc17343af52a841bca2a8bad7e5e")
+
+        has_update, info = download_manager.send(:check_update)
+        expect(has_update).to be_falsey
+      end
+
+      it "should return true when md5 does not match" do
+        expect(mock_metadata).to receive(:gz_md5).and_return("bca2a8bad7e5e4013dc17343af52a841")
+
+        has_update, info = download_manager.send(:check_update)
+        expect(has_update).to be_truthy
+      end
+    end
+
+    context "download database" do
+      let(:db_info) do
+        {
+          "md5_hash" => md5_hash,
+          "name" => filename,
+          "provider" => "maxmind",
+          "updated" => 1609891257,
+          "url" => "https://github.com/logstash-plugins/logstash-filter-geoip/archive/master.zip"
+        }
+      end
+      let(:md5_hash) { SecureRandom.hex }
+      let(:filename) { "GeoLite2-City.mmdb.gz"}
+
+      it "should raise error if md5 does not match" do
+        allow(Down).to receive(:download)
+        expect{ download_manager.send(:download_database, db_info) }.to raise_error /wrong checksum/
+      end
+
+      it "should download file and return zip path" do
+        expect(download_manager).to receive(:md5).and_return(md5_hash)
+
+        path = download_manager.send(:download_database, db_info)
+        expect(path).to match /GeoLite2-City_\d+\.mmdb\.gz/
+        expect(::File.exist?(path)).to be_truthy
+        ::File.delete(path) if ::File.exist?(path)
+      end
+    end
+
+    context "unzip" do
+      before(:each) do
+        file_path = ::File.expand_path("./fixtures/sample", ::File.dirname(__FILE__))
+        ::File.delete(file_path) if ::File.exist?(file_path)
+      end
+
+      it "gz file" do
+        path = ::File.expand_path("./fixtures/sample.gz", ::File.dirname(__FILE__))
+        unzip_path = download_manager.send(:unzip, path)
+        expect(::File.exist?(unzip_path)).to be_truthy
+      end
+    end
+
+    # context "assert database" do
```

Comment:
once the latest `logstash-filter-geoip` is published, these tests which point to [database_valid?](https://github.com/logstash-plugins/logstash-filter-geoip/pull/172/files#diff-4d7b5cc1053144fe24d3dd0c93195d93fa7d43e87ba683580f8d0b9b093cb5b0R79) can run again.